backend/src/db.rs

```diff
@@ -253,30 +253,46 @@ impl DB {
     }
 
     #[instrument(err, skip(self))]
-    pub async fn create_bot(&self, owner_id: &str, bot_name: &str) -> SqlxResult<User> {
+    pub async fn create_bot(&self, owner_id: &str, bot_name: &str) -> SqlxResult<CreateBotStatus> {
         let bot_id = uuid::Uuid::new_v4().to_string();
         let bot_name = format!("bot:{bot_name}");
         let mut transaction = self.pool.begin().await?;
-        sqlx::query!(
+        let result = sqlx::query!(
             r#"INSERT INTO user (id, name, balance, is_bot) VALUES (?, ?, '0', TRUE)"#,
             bot_id,
             bot_name
         )
         .execute(transaction.as_mut())
-        .await?;
-        sqlx::query!(
-            r#"INSERT INTO bot_owner (owner_id, bot_id) VALUES (?, ?)"#,
-            owner_id,
-            bot_id
-        )
-        .execute(transaction.as_mut())
-        .await?;
-        transaction.commit().await?;
-        Ok(User {
-            id: bot_id,
-            name: bot_name,
-            is_bot: true,
-        })
+        .await;
+
+        match result {
```

Comment:
I would probably do `if let Err(e) = result` rather than adding two indent levels to everything.